src/jinja2/compiler.py

```diff
@@ -1167,18 +1193,38 @@ def visit_If(self, node, frame):
         self.visit(node.test, if_frame)
         self.write(":")
         self.indent()
+
+        # check during runtime that filters used inside of the executed
+        # block are defined, as this step was skipped during compile time
+        def check_filters(node):
+            for filter in node.find_all(nodes.Filter, exclude=nodes.If):
+                self.writeline(f"if {self.filters[filter.name]} is None:")
```

Comment:
Should this use `get`?

Reply:
i saw  `__getitem__` used here, but it could be `get` in case https://github.com/pallets/jinja/blob/a127d9f452da9f6bfd2f7295cec907bc8399259f/src/jinja2/compiler.py#L1685-L1688